windows/wrapper/impl_webrtc_VideoCapturer.cpp

```diff
@@ -1122,51 +1121,71 @@ namespace webrtc
 #endif
 
       // Find a video profile by unique ID (if requested) or pick one with a
-      // resolution matching the resolution filters if only a kind was specified.
-      for (auto&& profile : profiles) {
-        if (profile.Id() == video_profile_id_utf16) {
-          video_profile_id_ = video_profile_id_utf16;
-          break;
-        }
-        else if (video_profile_id_utf16.empty()) {
+      // resolution matching the resolution filters if only a kind was
+      // specified.
+      if (video_profile_id_utf16.empty()) {
+        // Use video profile with closest capture format.
+        for (auto&& profile : profiles) {
           auto descs = profile.SupportedRecordMediaDescription();
           bool found = false;
+          double bestDeltaFramerate = 1.0;  // min non-acceptable value
           for (auto&& desc : descs) {
+            // Match resolution by exact width/height pixel value
             if ((width > 0) && (desc.Width() != (uint32_t)width)) {
               continue;
             }
             if ((height > 0) && (desc.Height() != (uint32_t)height)) {
               continue;
             }
-            if ((framerate > 0) && (desc.FrameRate() != framerate)) {
-              continue;
+            // Match framerate by closest match within a limit of 1 FPS
+            if (framerate > 0) {
+              double curFramerate = desc.FrameRate();
+              double deltaFramerate = std::fabs(curFramerate - framerate);
+              if (deltaFramerate >= bestDeltaFramerate) {
```

Comment:
>deltaFramerate >= bestDeltaFramerate [](start = 18, length = 36)

Please compare using epsilon.

Reply:
This is a best-fit algorithm, using epsilon will not select the best candidate.